src/test/regress/sql/multi_schema_support.sql

```diff
@@ -69,3 +70,301 @@ SELECT master_append_table_to_shard(1190001, 'nation_local', 'localhost', :maste
+
+-- create shard with master_create_worker_shards
+CREATE TABLE test_schema_support.nation_hash(
+    n_nationkey integer not null,
+    n_name char(25) not null,
+    n_regionkey integer not null,
+    n_comment varchar(152)
+);
+SELECT master_create_distributed_table('test_schema_support.nation_hash', 'n_nationkey', 'hash');
+SELECT master_create_worker_shards('test_schema_support.nation_hash', 4, 1);
+
+
+-- test cursors
+SET search_path TO public;
+BEGIN;
+DECLARE test_cursor CURSOR FOR 
+    SELECT *
+        FROM test_schema_support.nation_append
+        WHERE n_nationkey = 1;
+        FETCH test_cursor;
+FETCH test_cursor;
+END;
+
+-- test with search_path is set
+SET search_path TO test_schema_support;
+BEGIN;
+DECLARE test_cursor CURSOR FOR 
+    SELECT *
+        FROM nation_append
+        WHERE n_nationkey = 1;
+        FETCH test_cursor;
+FETCH test_cursor;
+END;
+
+
+-- test inserting to table in different schema
+SET search_path TO public;
+
+INSERT INTO test_schema_support.nation_hash(n_nationkey, n_name, n_regionkey) VALUES (6, 'FRANCE', 3);
+
+-- verify insertion
+SELECT * FROM test_schema_support.nation_hash WHERE n_nationkey = 6;
+
+-- test with search_path is set
+SET search_path TO test_schema_support;
+
+INSERT INTO nation_hash(n_nationkey, n_name, n_regionkey) VALUES (7, 'GERMANY', 3);
+
+-- verify insertion
+SELECT * FROM nation_hash WHERE n_nationkey = 7;
+
+
+-- test UDFs with schemas
+SET search_path TO public;
+
+\COPY test_schema_support.nation_hash FROM STDIN with delimiter '|';
+0|ALGERIA|0| haggle. carefully final deposits detect slyly agai
+1|ARGENTINA|1|al foxes promise slyly according to the regular accounts. bold requests alon
+2|BRAZIL|1|y alongside of the pending deposits. carefully special packages are about the ironic forges. slyly special 
+3|CANADA|1|eas hang ironic, silent packages. slyly regular packages are furiously over the tithes. fluffily bold
+4|EGYPT|4|y above the carefully unusual theodolites. final dugouts are quickly across the furiously regular d
+5|ETHIOPIA|0|ven packages wake quickly. regu
+\.
+
+-- create UDF in master node
+CREATE OR REPLACE FUNCTION dummyFunction(theValue integer)
+    RETURNS text AS
+$$
+DECLARE
+    strresult text;
+BEGIN
+    RETURN theValue * 3 / 2 + 1;
+END;
+$$
+LANGUAGE 'plpgsql' IMMUTABLE;
+
+-- create UDF in worker node 1
+\c - - - :worker_1_port
+CREATE OR REPLACE FUNCTION dummyFunction(theValue integer)
+    RETURNS text AS
+$$
+DECLARE
+    strresult text;
+BEGIN
+    RETURN theValue * 3 / 2 + 1;
+END;
+$$
+LANGUAGE 'plpgsql' IMMUTABLE;
+
+-- create UDF in worker node 2
+\c - - - :worker_2_port
+CREATE OR REPLACE FUNCTION dummyFunction(theValue integer)
+    RETURNS text AS
+$$
+DECLARE
+    strresult text;
+BEGIN
+    RETURN theValue * 3 / 2 + 1;
+END;
+$$
+LANGUAGE 'plpgsql' IMMUTABLE;
+
+\c - - - :master_port
+
+-- UDF in public, table in schema, search_path is not set
+SELECT dummyFunction(n_nationkey) FROM test_schema_support.nation_hash GROUP BY 1;
```

Comment:
Could you also add "order by"  to this query? The result does not need to be in the same order always.